File: 23_08_2024_Left_View_Of_Binary_Tree.java


//User function Template for Java

/* A Binary Tree node
class Node
{
    int data;
    Node left, right;

    Node(int item)
    {
        data = item;
        left = right = null;
    }
}*/
class Tree
{
    //Function to return list containing elements of left view of binary tree.
    ArrayList<Integer> leftView(Node root)
    {
      // Your code here
      ArrayList<Integer> list = new ArrayList<>();
      helper(root, 1, list);
      return list;
    }
    
    private void helper(Node root, int lvl, ArrayList<Integer> list){
        if(root == null) return;
        if(lvl > list.size()) list.add(root.data);
        helper(root.left, lvl+1, list);
        helper(root.right, lvl+1, list);
    }
}

File: 23_08_2024_Left_View_Of_Binary_Tree.py
#User function Template for python3


'''
# Node Class:
class Node:
    def _init_(self,val):
        self.data = val
        self.left = None
        self.right = None
'''

#Function to return a list containing elements of left view of the binary tree.
def LeftView(root):
    stack=[]
    mpp={}
    stack.append((root,0))
    while stack:
        pair=stack.pop()
        node=pair[0]
        y=pair[1]
        if y not in mpp:
            mpp[y]=node.data
        if node.right: stack.append((node.right,y+1))
        if node.left: stack.append((node.left,y+1))
    res=[]
    res = [mpp[key] for key in sorted(mpp)]
    return res
